### PR TITLE
Upgrade to RxJava 1.1.9 and add missing operators

### PR DESCRIPTION
**Pull request by zsxwing**

### CODE REVIEW

File: examples/src/test/scala/examples/TestSchedulerExample.scala [hunk 1]
```diff
@@ -21,7 +21,7 @@ import org.mockito.Mockito._
 import org.scalatest.junit.JUnitSuite
 import rx.lang.scala._
 import rx.lang.scala.schedulers.TestScheduler
-import rx.observers.TestObserver
```
Comment (zsxwing) [inline examples/src/test/scala/examples/TestSchedulerExample.scala:24]: `TestObserver` is deprecated.

File: src/main/scala/rx/lang/scala/observers/TestSubscriber.scala [hunk 2]
```diff
@@ -16,7 +16,7 @@
 package rx.lang.scala.observers
 
 import java.util.concurrent.TimeUnit
-import scala.collection.JavaConversions._
```
Comment (zsxwing) [inline src/main/scala/rx/lang/scala/observers/TestSubscriber.scala:19]: `JavaConversions` is deprecated in Scala 2.12.0-M5

File: src/main/scala/rx/lang/scala/observers/TestSubscriber.scala [hunk 3]
```diff
@@ -259,6 +259,40 @@ class TestSubscriber[T] private[scala](jTestSubscriber: JTestSubscriber[T]) exte
...
+
+  /**
+   * $experimental Returns the number of times onCompleted was called on this [[TestSubscriber]].
+   *
+   * @return the number of times onCompleted was called on this [[TestSubscriber]].
+   */
+  @Experimental
+  def getCompletions: Int = {
+    jTestSubscriber.getCompletions
+  }
```
Comment (samuelgruetter) [inline src/main/scala/rx/lang/scala/observers/TestSubscriber.scala:295]: `getXxx` method names are not so Scala-typical... Why not just `completions`?

File: build.sbt [hunk 4]
```diff
@@ -14,15 +14,19 @@ scalacOptions in ThisBuild := Seq("-feature", "-unchecked", "-deprecation", "-en
...
 parallelExecution in Test := false
 
-libraryDependencies ++= Seq(
-  "io.reactivex" % "rxjava" % "1.1.6",
-  "org.mockito" % "mockito-core" % "1.9.5" % "test",
-  "junit" % "junit" % "4.11" % "test",
-  "org.scalatest" %% "scalatest" % "2.2.6" % "test")
+libraryDependencies ++= {
+  // Scalatest 3.0.0 supports 2.12.0-M5 but it drops Scala 2.10 support. So just use 2.2.6 for Scala 2.10.
+  val scalatestVersion = if (scalaVersion.value.startsWith("2.10")) "2.2.6" else "3.0.0"
```
Comment (samuelgruetter) [inline build.sbt:23]: nice :smiley:

**Comment by samuelgruetter** [comment]
LGTM :+1: 
thanks @imcharsi and @zsxwing

**Pull request merged**